jobsdb/jobsdb_test.go

```diff
@@ -169,3 +180,859 @@ var _ = Describe("Calculate newDSIdx for cluster migrations", func() {
+			stmt = c.mock.ExpectPrepare(fmt.Sprintf(`COPY "%s" ("job_id", "job_state", "attempt", "exec_time",
+			"retry_time", "error_code", "error_response") FROM STDIN`, ds.JobStatusTable))
+			status = statusList[1]
+			stmt.ExpectExec().WithArgs(status.JobID, status.JobState, status.AttemptNum, status.ExecTime,
+				status.RetryTime, status.ErrorCode, string(status.ErrorResponse)).WillReturnResult(sqlmock.NewResult(0, 1))
+			stmt.ExpectExec().WithArgs().WillReturnResult(sqlmock.NewResult(0, int64(len(properStoreJobs))))
+
+			c.mock.ExpectCommit()
+
+			err := jd.UpdateJobStatus(statusList, []string{"GW"}, nil)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// we make sure that all expectations were met
+			if err := c.mock.ExpectationsWereMet(); err != nil {
+				ginkgo.Fail("there were unfulfilled expectations")
+			}
+		})
+
+		It("should fail to update job status", func() {
+			c.mock.ExpectBegin()
+
+			ds := dsListInMemory[0]
+			stmt := c.mock.ExpectPrepare(fmt.Sprintf(`COPY "%s" ("job_id", "job_state", "attempt", "exec_time",
+			"retry_time", "error_code", "error_response") FROM STDIN`, ds.JobStatusTable))
+			status := statusList[0]
+			stmt.ExpectExec().WithArgs(status.JobID, status.JobState, status.AttemptNum, status.ExecTime,
+				status.RetryTime, status.ErrorCode, string(status.ErrorResponse)).WillReturnResult(sqlmock.NewResult(0, 1))
+			stmt.ExpectExec().WithArgs().WillReturnResult(sqlmock.NewResult(0, int64(len(properStoreJobs))))
+
+			ds = dsListInMemory[1]
+			stmt = c.mock.ExpectPrepare(fmt.Sprintf(`COPY "%s" ("job_id", "job_state", "attempt", "exec_time",
+			"retry_time", "error_code", "error_response") FROM STDIN`, ds.JobStatusTable))
+			status = statusList[1]
+			stmt.ExpectExec().WithArgs(status.JobID, status.JobState, status.AttemptNum, status.ExecTime,
+				status.RetryTime, status.ErrorCode, string(status.ErrorResponse)).WillReturnResult(sqlmock.NewResult(0, 1))
+			stmt.ExpectExec().WithArgs().WillReturnError(errors.New("exec failed"))
+
+			c.mock.ExpectRollback()
+
+			err := jd.UpdateJobStatus(statusList, []string{"GW"}, nil)
+			Expect(err).To(Equal(errors.New("exec failed")))
+
+			// we make sure that all expectations were met
+			if err := c.mock.ExpectationsWereMet(); err != nil {
+				ginkgo.Fail("there were unfulfilled expectations")
+			}
+		})
+	})
+
+	Context("GetProcesed", func() {
+		var jd *HandleT
+
+		BeforeEach(func() {
+			jd = &HandleT{}
+			jd.dbHandle = c.db
+			jd.datasetList = dsListInMemory
+			jd.datasetRangeList = dsRangeList
+			jd.enableWriterQueue = true
+			jd.workersAndAuxSetup(ReadWrite, "tt", 0*time.Hour, "", false, QueryFiltersT{})
+		})
+
+		assertGetProcessedJobsWithCustomVal := func(state string) {
+			It("should return jobs with non terminal last state when queried with customval", func() {
+				var stateQuery, customValQuery, sourceQuery, limitQuery string
+				stateQuery = " AND ((job_state='failed'))"
+				customValQuery = " AND ((tt_jobs_1.custom_val='GW'))"
```

Comment:
Do we need to hardcode the custom val to 'GW' or can we something like 'some-ds-name'